client/src/components/DatasetInformation/DatasetDetails.vue

```diff
@@ -41,42 +41,43 @@ const datasetLoadingError = ref<string | null>(null);
 
 async function getDatasetDetails() {
     loading.value = true;
-
     try {
         const data = await fetchDatasetDetails({ id: props.datasetId });
-
         dataset.value = data;
     } catch (e) {
-        const error = e as AxiosError<{ err_msg?: string }>;
-
-        datasetLoadingError.value = error.response?.data?.err_msg || "Unable to fetch available dataset details.";
+        datasetLoadingError.value = errorMessageAsString(e) || "Unable to fetch available dataset details.";
     } finally {
         loading.value = false;
     }
 }
 
 async function loadJobDetails() {
-    try {
-        const { data } = await fetchJobDetails({ job_id: dataset.value?.creating_job as string, full: true });
-
-        if (stateIsTerminal(data)) {
-            clearTimeout(jobTimeOut.value);
-        } else {
-            jobTimeOut.value = setTimeout(loadJobDetails, 3000);
-        }
-
-        jobDetails.value = data;
-    } catch (e) {
-        const error = e as AxiosError<{ err_msg?: string }>;
+    const { data, error } = await GalaxyApi().GET("/api/jobs/{job_id}", {
+        params: {
+            path: { job_id: dataset.value?.creating_job! },
+            query: { full: true },
+        },
+    });
+
+    if (error) {
+        jobLoadingError.value = errorMessageAsString(error);
```

Comment:
This seems like a better example to me @davelopez. Toasts are simple, but they often lack context, especially if there's more than one possible error on a page.

Reply:
Done! https://github.com/galaxyproject/galaxy/pull/18532/commits/b8be283f6ee3a6eb08186f83acd6ffc4fcae7458